rules.py

```diff
@@ -218,6 +223,73 @@ def check(self, element):
                 "Line %d. %s is empty"% (
                     element.sourceline, element.tag))
 
+
+class DuplicateID(base.TreeRule):
+    """Check that the file does not contain duplicate object IDs
+    """
+    def check(self):
+        all_object_ids = set()
+        error_log = []
+        for event, element in etree.iterwalk(
+                self.election_tree, events=("end",)):
+            if "objectId" not in element.attrib:
+                continue
+            else:
+                obj_id = element.get("objectId")
+                if not obj_id:
+                    continue
+                if obj_id in all_object_ids:
+                    error_line = element.sourceline
+                    error_message = "{0} is a duplicate object ID".format(
+                        obj_id)
+                    error_log.append(base.ErrorLogEntry(
+                        error_line, error_message))
+                else:
+                    all_object_ids.add(obj_id)
+        if error_log:
+            raise base.ElectionTreeError(
+                "The Election File contains duplicate object IDs", error_log)
+
+
+class ValidIDREF(base.BaseRule):
+    """Check that every field of type IDREF actually references a value that
+    exists in a field of type ID.
+    """
+
+    all_object_ids = set()
+
+    def __init__(self, election_tree, schema_file):
+        super(ValidIDREF, self).__init__(election_tree, schema_file)
+        for event, element in etree.iterwalk(
+                self.election_tree, events=("end",)):
+            if "objectId" not in element.attrib:
+                continue
+            else:
+                obj_id = element.get("objectId")
+                if not obj_id:
+                    continue
+                else:
+                    self.all_object_ids.add(obj_id)
+
+    def elements(self):
+        schema_tree = etree.parse(self.schema_file)
+        eligible_elements = []
+        for event, element in etree.iterwalk(schema_tree):
+            tag = self.strip_schema_ns(element)
+            if (tag and tag == "element" and
+                    element.get("type") in ("xs:IDREF", "xs:IDREFS")):
+                eligible_elements.append(element.get("name"))
+        return eligible_elements
+
+    def check(self, element):
+        if element.text:
+            id_references = element.text.split(" ")
```

Comment:
Change split(" ") to split(). It'll handle trailing whitespace as well as multiple intermediate whitespace characters correctly.